src/internal/connector/graph/concurrency_limiter_test.go

```diff
@@ -0,0 +1,117 @@
+package graph
+
+import (
+	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+
+	khttp "github.com/microsoft/kiota-http-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/tester"
+)
+
+type ConcurrencyLimiterUnitTestSuite struct {
+	tester.Suite
+}
+
+func TestConcurrencyLimiterSuite(t *testing.T) {
+	suite.Run(t, &ConcurrencyLimiterUnitTestSuite{Suite: tester.NewUnitSuite(t)})
+}
+
+func (suite *ConcurrencyLimiterUnitTestSuite) TestConcurrencyLimiter() {
+	t := suite.T()
+
+	maxConcurrentRequests := 4
+	cl := generateConcurrencyLimiter(maxConcurrentRequests)
+	client := khttp.GetDefaultClient(cl)
+
+	// Server side handler to simulate 429s
+	sem := make(chan struct{}, maxConcurrentRequests)
+	reqHandler := func(w http.ResponseWriter, r *http.Request) {
+		select {
+		case sem <- struct{}{}:
+			defer func() {
+				<-sem
+			}()
+
+			time.Sleep(time.Duration(rand.Intn(50)+50) * time.Millisecond)
+			w.WriteHeader(http.StatusOK)
+
+			return
+		default:
+			w.WriteHeader(http.StatusTooManyRequests)
```

Comment:
purely as a sanity check to ensure the test is running as expected, how about adding an incremental counter to this default block, and asserting that it's non-zero after the wg.Wait()?

Reply:
If the concurrency limiter semaphore capacity <= reqHandler semaphore capacity, we would never hit the default block. 

I've manually tested that the reqHandler default block is working as expected. I could add another test which hits the default block, but then that'd be testing the test handler. Not sure if there is value in doing that.